### PR TITLE
Add guesses to `linearize`

### PR DESCRIPTION
**Pull request by ven-k**
Adds the guesses of the sys to x0 (Preference = guess < defaults < op.)

While here, fixes the metadata in mtkmodel `@defaults` for parameters of subcomponents.

## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

File: src/systems/abstractsystem.jl [hunk 1]
```diff
@@ -1997,7 +1997,7 @@ lsys_sym, _ = ModelingToolkit.linearize_symbolic(cl, [f.u], [p.x])
 """
 function linearize(sys, lin_fun; t = 0.0, op = Dict(), allow_input_derivatives = false,
         p = DiffEqBase.NullParameters())
-    x0 = merge(defaults(sys), op)
+    x0 = merge(defaults(sys), Dict(missing_variable_defaults(sys)), op)
```
Comment (ven-k) [inline src/systems/abstractsystem.jl:2000]: I've matched with `linearization_function`, which doesn't account for guesses either.

Should both be instead
```julia
x0 = merge(defaults(sys), get_guesses(sys), Dict(missing_variable_defaults(sys)), op)
```

File: test/downstream/linearize.jl [hunk 2]
```diff
@@ -19,6 +19,24 @@ lsys, ssys = linearize(sys, [r], [y])
...
+@parameters kp = 1
+D = Differential(t)
+
+eqs = [u ~ kp * (r - y)
+       D(x) ~ -x + u
+       y ~ x]
+
+@named sys = ODESystem(eqs, t)
+
+lsys, ssys = linearize(sys, [r], [y])
```
Comment (ChrisRackauckas) [inline test/downstream/linearize.jl:33]: This should error as most of the values are not defined. It's not clear what to linearize at.

File: test/downstream/linearize.jl [hunk 3]
```diff
@@ -19,6 +19,24 @@ lsys, ssys = linearize(sys, [r], [y])
 @test lsys.C[] == 1
 @test lsys.D[] == 0
 
+# With guesses
+@variables t x(t) [guess = 0] y(t) [guess = 0] u(t) [guess = 0] r(t)=0.0 [input = true]
```
Comment (ChrisRackauckas) [inline test/downstream/linearize.jl:23]:
```suggestion
@variables t x(t) [guess = 0] y(t) [guess = 1] u(t) [guess = 0] r(t)=0.0 [input = true]
```

This is a much better test case which then highlights the issue with your proposed approach.

Comment (ven-k) [reply test/downstream/linearize.jl:23]: Yes; will add non zero guesses.

**Pull request closed without merging**